src/backend/marsha/core/api.py

```diff
@@ -678,6 +686,110 @@ def initiate_upload(self, request, pk=None):
         return Response(response)
 
 
+class LiveRegistrationViewSet(
+    ObjectPkMixin,
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+):
+    """Viewset for the API of the LiveRegistration object."""
+
+    permission_classes = [permissions.IsVideoToken]
+    queryset = LiveRegistration.objects.all()
+    serializer_class = serializers.LiveRegistrationSerializer
+
+    def get_queryset(self):
+        """Restrict list access to liveRegistration related to the video in the JWT token."""
+        user = self.request.user
+        if isinstance(user, TokenUser) and user.token.payload is not None:
+            # admin and instructors can access all registrations
+            if user.token.payload.get("roles", None) is not None and any(
+                role in ["administrator", "instructor"]
+                for role in user.token.payload["roles"]
+            ):
+                return LiveRegistration.objects.filter(
+                    id=self.kwargs["pk"], video__id=user.id
+                )
+            # others can only read their registrations
+            if user.token.payload.get("user", None) is not None:
+                if user.token.payload["user"].get("email", None) is not None:
+                    # token has an email
+                    return LiveRegistration.objects.filter(
+                        id=self.kwargs["pk"],
+                        video__id=user.id,
+                        email=user.token.payload["user"]["email"],
+                    )
+                # no email, user has access to this registration if it's the right combinaison of
+                # lti_user_id and lti_consumer_id
+                if (
+                    user.token.payload["user"].get("id", None) is not None
+                    and user.token.payload.get("context_id", None) is not None
+                ):
+                    playlist = Playlist.objects.get(
+                        lti_id=user.token.payload["context_id"]
+                    )
+                    return LiveRegistration.objects.filter(
+                        id=self.kwargs["pk"],
+                        lti_consumer_site=playlist.consumer_site,
+                        lti_user_id=user.token.payload["user"]["id"],
+                        video__id=user.id,
+                    )
+
+        return LiveRegistration.objects.none()
+
+    def perform_create(self, serializer):
+        """Use data from token before saving."""
+        user = self.request.user
+        if user.token.payload.get("user", None) is not None:
+            email = self.request.data["email"]
+            lti_user_id = self.request.data.get("lti_user_id", None)
+            lti_consumer_site = self.request.data.get("lti_consumer_site", None)
+            # if email is present in token, it's assigned to data, value sent gets ignored
+            if user.token.payload["user"].get("email", None) is not None:
```

Comment:
I don't remember we decided this behaviour. I thought the user will be free to choose the registered email.

Reply:
In the front the user that has an email in its token won't be able to choose its email, it will be automatically is one. The input will only be for the others, it's what we had decided in my knowledge anyway :-)

Reply:
We discussed 2 scenarios:

- the email input will be prefilled with the email and the user can change it.
- the input is not displayed but in that case (for what I understand) it's an hidden field. 

In both cases the email is sent in the request.
 
IMO the email should be set in the request and not picked in the token in the back application.

Reply:
Why not secure the email from the token which is the one expected ?

Reply:
Because in the first scenario the user can choose an other email than the one he is registered with.

Reply:
ah ok, so no, if the user has an email he won't be able to change it, he will only be able to register, we're only going to ask the email for the one that don't have any.

Reply:
As discussed during the stand-up, we now generate a 401 if it's not the expected email, so this way we use data from the query and at the same time control it.

Reply:
a `400` not a `401`